http/oidc/src/main/java/org/wildfly/security/http/oidc/OidcRequestAuthenticator.java

```diff
@@ -456,6 +459,14 @@ protected AuthChallenge resolveCode(String code) {
             log.error("Stale token");
             return challenge(HttpStatus.SC_FORBIDDEN, AuthenticationError.Reason.STALE_TOKEN, null);
         }
+
+        String stateCookieValue = getCookieValue(deployment.getStateCookieName());
```

Comment:
A separate cookie should be used for nonce purposes.

Reply:
done